components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.probes/src/main/java/org/wso2/micro/integrator/probes/ReadinessResource.java

```diff
@@ -78,33 +77,25 @@ public boolean invoke(MessageContext synCtx) {
 
         String response = "";
 
-        if (cachedResponse.isEmpty()) {
-            // appending MI server version number to the response
-            CarbonServerConfigurationService serverConfig = CarbonServerConfigurationService.getInstance();
-            String miVersion = serverConfig.getServerVersion();
-            response = "{\"version\":\"" + miVersion + "\",";
+        // appending MI server version number to the response
+        CarbonServerConfigurationService serverConfig = CarbonServerConfigurationService.getInstance();
+        String miVersion = serverConfig.getServerVersion();
+        response = "{\"version\":\"" + miVersion + "\",";
 
-            if (!isHotDeploymentEnabled) {
-                ArrayList<String> faultyCapps = new ArrayList<>(CappDeployer.getFaultyCapps());
-                if (!faultyCapps.isEmpty()) {
-                    String faultyList = String.join("\",\"", faultyCapps);
-                    response += "\"status\": \"not ready, faulty CAPPs detected\", \"Faulty CAPPs\" : [\"" + faultyList
-                            + "\"]}";
-                    cachedResponseCode = 500;
-                } else {
-                    response += "\"status\" : \"ready\"}";
-                    cachedResponseCode = 200;
-                }
-            } else {
-                response += "\"status\" : \"ready\"}";
-                cachedResponseCode = 200;
-            }
-            // update cached response
-            cachedResponse = response;
+        ArrayList<String> faultyCapps = new ArrayList<>(CappDeployer.getFaultyCapps());
+        if (!faultyCapps.isEmpty()) {
+            String faultyList = String.join("\",\"", faultyCapps);
+            response += "\"status\": \"not ready, faulty CAPPs detected\", \"Faulty CAPPs\" : [\"" + faultyList
+                    + "\"]}";
+            responseCode = 500;
+        } else {
+            response += "\"status\" : \"ready\"}";
```

Comment:
how do we ensure that deployment is completed by this time

Reply:
> There were test cases for this, did you happen to check them too with the fix

Removed the invalid tests

Reply:
> how do we ensure that deployment is completed by this time

https://ei.docs.wso2.com/en/7.2.0/micro-integrator/setup/deployment/health_check/
quote "Initial delay and the period has to be fine-tuned according to your deployment."